### PR TITLE
pythonPackages.nose-randomly: fix tests on Darwin

### PR DESCRIPTION
**Pull request by austinbutler**
###### Motivation for this change

Similar to https://github.com/NixOS/nixpkgs/pull/145276 and a few others before that, the tmp path is too long on macOS.

I'd open an issue upstream, but the repo is archived.

###### Things done

- Built on platform(s)
  - [x] x86_64-linux
  - [ ] aarch64-linux
  - [x] x86_64-darwin
  - [ ] aarch64-darwin
- [ ] For non-Linux: Is `sandbox = true` set in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
- [x] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://nixos.org/manual/nixpkgs/unstable/#sec-package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [ ] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [22.05 Release Notes (or backporting 21.11 Release notes)](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md#generating-2205-release-notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
  - [ ] (Release notes changes) Ran `nixos/doc/manual/md-to-db.sh` to update generated release notes
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

### CODE REVIEW

File: pkgs/development/python-modules/nose-randomly/default.nix [hunk 1]
```diff
@@ -16,9 +17,10 @@ buildPythonPackage rec {
 
   checkInputs = [ numpy nose ];
 
-  checkPhase = ''
-    nosetests
-  '';
+  checkPhase = if stdenv.isDarwin then ''
+    # Work around "OSError: AF_UNIX path too long"
+    TMPDIR="/tmp" nosetests
```
Comment (SuperSandro2000) [inline pkgs/development/python-modules/nose-randomly/default.nix:22]: Could we do ``export TMPDIR=$(mktemp -d)`` and set that for everyone?

Also please target master. The rebuild could is not that high.

Comment (austinbutler) [reply pkgs/development/python-modules/nose-randomly/default.nix:22]: `mktemp -d` has come up several times before, see https://github.com/NixOS/nixpkgs/pull/126642#discussion_r650556767 and https://github.com/NixOS/nixpkgs/pull/145276#discussion_r746458026.

I will change target to master.

Comment (austinbutler) [reply pkgs/development/python-modules/nose-randomly/default.nix:22]: Updated.

File: pkgs/development/python-modules/nose-randomly/default.nix [hunk 2]
```diff
@@ -16,9 +17,10 @@ buildPythonPackage rec {
...
+  '' else "nosetests";
```
Comment (SuperSandro2000) [inline pkgs/development/python-modules/nose-randomly/default.nix:23]:
```suggestion
  '' else ''
    nosetests
  '';
```
That should avoid the rebuilds on linux

Comment (austinbutler) [reply pkgs/development/python-modules/nose-randomly/default.nix]: TIL! Do you want me to squash back to one commit?

Comment (SuperSandro2000) [reply pkgs/development/python-modules/nose-randomly/default.nix]: I just squash merged.

**Pull request merged**